src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T[]): GroupsMap<T> {
+  const keys: string| number[] = [];
+
+  items.map((item) => (keys.includes(item[key])
+    ? null
+    : keys.push(item[key])));
+
+  const result = {};
+
+  keys.map((k) => (result[k] = items.filter((el) => el[key] === k)));
```

Comment:
same here, better to  use the reduce

Reply:
I can't think of how to use reduce here

Reply:
@YuriiHlushenko try to rewrite it with reduce when u have time)